tao/tao_bridge/kernels/tao_compilation_cache.cc

```diff
@@ -316,6 +316,221 @@ Status GetDeviceId(OpKernelContext* ctx, int* device_id) {
   return ok ? Status::OK() : errors::Internal("Could not parse device name.");
 }
 
+using ItemSignature = TaoCompilationCache::Signature;
+
+// A class used to load/store compliatoin cache on disc.
+// Note that:
+// 1, we suppose the clustering result is stable across different process.
+// The above assumption may not always hold. Thus be careful when using this
+// feature. 2, We may have multiple intances of TaoCompilationCache if we have
+// multiple sessions in the same process. However there should be only one
+// instance of PersistentCompliationCache in order to get stable result.
+class PersistentCompliationCache {
+ public:
+  // cache_dump_path: the path used to store the cache files.
+  explicit PersistentCompliationCache(const string& cache_dump_path)
+      : cache_dump_path_(cache_dump_path) {
+    auto status = LoadFromFile();
+    TF_CHECK_OK(status);
+  }
+
+  ~PersistentCompliationCache() {
+    auto status = DumpToFile();
+    if (!status.ok()) {
+      VLOG(0) << "[WARNING] fail to dump compilation cache: "
+              << status.error_message();
+    }
+  }
+
+  // Returns the name of file used to store `CompilationCacheResult`.
+  static string getCacheTableFileName() { return "disc_cache"; }
+
+  // Returns the path of file used to store `CompilationCacheResult`.
+  string getCacheTableFilePath() {
+    return absl::StrCat(cache_dump_path_, "/", getCacheTableFileName());
+  }
+
+  // Returns the singleton
+  static PersistentCompliationCache& Global() {
```

Comment:
How about removing copy ctor of `PersistentCompliationCache` to prevent miss using like:
```
PersistentCompliationCache cache = PersistentCompliationCache::Global()
```

Reply:
done, thanks.